src/vs/workbench/parts/output/common/outputServices.ts

```diff
@@ -155,44 +145,36 @@ export class OutputService implements IOutputService {
 		return Object.keys(this.receivedOutput);
 	}
 
-	public clearOutput(channel = DEFAULT_OUTPUT_CHANNEL): void {
-		this.receivedOutput[channel] = '';
+	public getActiveChannel(): string {
+		if (!this.outputPanel || !this.outputPanel.getInput()) {
+			return null;
+		}
 
-		this._onOutput.fire({ channel: channel, output: null /* indicator to clear output */ });
+		return (<OutputEditorInput>this.outputPanel.getInput()).getChannel();
 	}
 
-	public showOutput(channel: string = DEFAULT_OUTPUT_CHANNEL, sideBySide?: boolean | Position, preserveFocus?: boolean): TPromise<IEditor> {
-
-		// If already opened, focus it unless we want to preserve focus
-		let existingOutputEditor = this.findOutputEditor(channel);
-		if (existingOutputEditor) {
-			if (!preserveFocus) {
-				return this.editorService.focusEditor(existingOutputEditor);
-			}
-
-			// Still reveal last line
-			(<OutputEditorInput>existingOutputEditor.input).revealLastLine();
+	public clearOutput(channel: string): void {
+		this.receivedOutput[channel] = '';
 
-			return Promise.as(existingOutputEditor);
-		}
+		this._onOutput.fire({ channel: channel, output: null /* indicator to clear output */ });
+	}
 
-		// Otherwise open new
-		return this.editorService.openEditor(OutputEditorInput.getInstance(this.instantiationService, channel), preserveFocus ? EditorOptions.create({ preserveFocus: true }) : null, <any>sideBySide);
+	public showOutput(channel: string, preserveFocus?: boolean): TPromise<IEditor> {
+		return this.panelService.openPanel(OUTPUT_PANEL_ID, !preserveFocus).then((panel: OutputPanel) => {
+			this.outputPanel = panel;
```

Comment:
@isidorn this seems weird to me, imagine I bring up the panel by just toggling it and output happens to be the last panel I had in there. all your assumptions about having the outputPanel inside the service would fail because the output panel was not revealed through this service. idealy I would try to reduce the use of outputPanel inside the service to a minimum. do we need it in all cases? could the active channel also be a property on the service? could "revealLastLine" move out and someone that wants to reveal just has to open the output panel and call reveal on it? think towards that and maybe we can get rid of the "this.outputPanel" in the service completly!